ssl/ssl.go

```diff
@@ -16,6 +18,28 @@ func GetRootCAPool() *x509.CertPool {
 	return pool
 }
 
+// Appends certificates to the `x509.CertPool` from a `.pem` private local file. On many Linux
+// systems, /etc/ssl/cert.pem will contain the system wide set but in our case, we'll pull
+// the certificate file path from the `Configuration` struct
+func AppendPEMFileToRootCAPool(certPool *x509.CertPool, pemFileName string) (*x509.CertPool, error) {
+	if certPool == nil {
+		certPool = x509.NewCertPool()
+	}
+	if pemFileName != "" {
+		//read file and place it's contents in `pemCerts`
+		pemCerts, err := ioutil.ReadFile(pemFileName)
+		if err != nil {
+			return certPool, fmt.Errorf("Failed to read file %s: %v", pemFileName, err)
+		}
+
+		//`pemCerts` has been obtained, append to certPool
+		certPool.AppendCertsFromPEM(pemCerts)
+	} else {
+		return certPool, fmt.Errorf("Path to local PEM file containing SSL certificates is\n")
```

Comment:
This error message is confusing ... is it really an error to fail to read a cert file when none was given?

Reply:
Removed